auto_update_tests.py

```diff
@@ -152,7 +152,7 @@ def update_example_tests():
             os.remove(output_file)
             if sys.platform == 'darwin':
                 # Also removes debug directory produced on Mac OS
-                shutil.rmtree(output_file + '.dSYM')
+                shutil.rmtree(output_file + '.dSYM', ignore_errors=True)
```

Comment:
It looks like the similar change to check.py in #2646 was to check to existence instead.

However .. is this actually needed?  Can we just remove these two lines?  I'm not macOS users so I don't know when or why this file might get created.

Reply:
I totally agree for removing this. I was just not sure if it was used for something.

Reply:
If we go that route could you remove the same code from check.py too?    Before we that though it would be good to understand why it was ever needed and under what circumstances calling `g++` alone could generate this directory.    

if you want to land this now and just match the code in check.py I'm ok with that too.

Reply:
sure. Let me try